src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -2812,8 +2824,14 @@ def remove_cors(cmd, resource_group_name, name, allowed_origins, slot=None):
             configs.cors.allowed_origins = [x for x in (configs.cors.allowed_origins or []) if x not in allowed_origins]
         else:
             configs.cors.allowed_origins = []
-        configs = _generic_site_operation(cmd.cli_ctx, resource_group_name, name, 'update_configuration', slot, configs)
-    return configs.cors
+
+    result = None
+    if is_centauri_functionapp(cmd, resource_group_name, name):
+        result = update_configuration_polling(cmd, resource_group_name, name, slot, configs)
+    else:
+        result = _generic_site_operation(cmd.cli_ctx, resource_group_name, name, 'update_configuration', slot, configs)
+
+    return result.cors
```

Comment:
I suspect this might lead to issues. The rest of the function's code is only run if `configs.cors` is not None, so I think this logic should only run in that case as well.